kedro_azureml/hooks.py

```diff
@@ -16,6 +16,13 @@ def after_context_created(self, context) -> None:
             )
         self.azure_config = AzureMLConfig(**context.config_loader["azureml"]["azure"])
 
+    @hook_impl
+    def after_catalog_created(self, catalog):
+        for dataset_name, dataset in catalog._data_sets.items():
+            if isinstance(dataset, AzureMLAssetDataSet):
```

Comment:
Could you explain the logic behind this hook?

Reply:
The idea is that we'll want to have the datasets to be downloadable by default. This allows them to be used in kedro ipython/jupyterlab or really any kedro sessions locally and not only in local pipeline runs. To achieve this the dataset needs access to the config to get the azureml config infos. The quickest way I could think of at this point is to use the `after_catalog_created` hook. Other options I thought of but dismissed was using the config_loader in the dataset or having the `config.json` created earlier in the session. Very open to ideas!

Reply:
Got it! In that case the `as_local` is misleading, as you're using it only to pass the `azure_config`. I would prefer to have this clearer as a separate method, because my first understanding of that code was "force this dataset to always be local" and then in the `before_pipeline_run` there's the `as_local` and `as_remote` again and it requires to know the whole reasoning behind this to understand the implementation.
If you've set the 
```
        self._download = True
        self._local_run = True
```
to be defaults, let's go with `dataset.azure_config = self.azure_config` in the `after_catalog_created` (`azure_config` as a `@property` with setter and getter in the dataset class).

Reply:
Nice, just add an entry in `CHANGELOG.md` and I will merge :)